2_segmenting_data/segmentation_utils.py

```diff
@@ -0,0 +1,251 @@
+
+    # CellPose model has many different parameters that can be found via the Command Line documentation (https://cellpose.readthedocs.io/en/latest/command.html), which
+    # can then be found for Python implementation using the CellPose GitHub (https://github.com/MouseLand/cellpose/tree/main/cellpose.)
+    cellpose_model = models.Cellpose(gpu=True, model_type=model_specs["model_type"])
+    masks, flows, styles, diams = cellpose_model.eval(
+        image,
+        diameter=model_specs["diameter"],
+        channels=model_specs["channels"],
+        flow_threshold=model_specs["flow_threshold"],
+    )
+
+    # Remove the cell masks that are on the edges of the image
+    if model_specs["remove_edge_masks"]:
+        masks = utils.remove_edge_masks(masks)
+
+    # CellPose will determine the (x,y) points that create the outlines for each of the cells being segmented. By determining the mean of the these outline points,
+    # we can then take the (x,y) center coordinates and add them into our objects data list from above
+    outlines = utils.outlines_list(masks)
+    for outline in outlines:
+        centroid = outline.mean(axis=0)
+        object_data = {
+            "Location_Center_X": centroid[0],
+            "Location_Center_Y": centroid[1],
+        }
+        objects_data.append(object_data)
+
+    # Covert the object data list into a pandas Dataframe
+    objects_data = pd.DataFrame(objects_data)
+    return objects_data
+
+
+def overlay_channels(identifier: str, images_dir: pathlib.Path):
+    """overlays cytoplasm, ER, and nuclei channels to help cytoplasm segmentation in CellPose
+
+    Args:
+        current_image (str): string for current site to overlay the channels for
+        current_dir (pathlib.Path): directory of where the current images used are located
+    """
+    # Takes the identifer of the nuclei channel and splits it into parts: well, plate, channel, site
+    # We then use all parts except for channel and change it for the GFP and RFP IDs
+    field_details = identifier.split("_")
+
+    dapi_id = identifier
+    gfp_id = f"{field_details[0]}_{field_details[1]}_2_{field_details[3]}"
+    rfp_id = f"{field_details[0]}_{field_details[1]}_3_{field_details[3]}"
+
+    # Iterate through the directory containing the images and if an image has the ID of the given channel, it will read in that image. There are three images total
+    # that are read in for each site that needs to be overlayed
+    for image_path in images_dir.iterdir():
+        if dapi_id in image_path.name:
+            dapi = io.imread(image_path)
+        if gfp_id in image_path.name:
+            gfp = io.imread(image_path)
+        if rfp_id in image_path.name:
+            rfp = io.imread(image_path)
+
+    # RFP = red channel (cytoplasm), GFP = green channel (ER), DAPI = blue (nuclei)
+    # Multiplier only assists in user interpretation of the images to configure model parameters in CellPose GUI
+    overlay = np.dstack([rfp * 3, gfp, dapi])
```

Comment:
Would it make sense to remove hardcoding of `3` and set a default argument of `rfp_channel_multiplier=3` as a function argument?

Reply:
This has been done!